cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/ClusterPartitionState.java

```diff
@@ -112,7 +113,12 @@ protected void populateKafkaPartitionState(Set<PartitionInfo> underReplicatedPar
         }
         for (PartitionInfo partitionInfo : _kafkaCluster.partitionsForTopic(topic)) {
           int numInsyncReplicas = partitionInfo.inSyncReplicas().length;
-          boolean isURP = !(partitionInfo.inSyncReplicas() == partitionInfo.replicas());
+          Node[] inSyncReplicas = Arrays.copyOf(partitionInfo.inSyncReplicas(), numInsyncReplicas);
+          Arrays.sort(inSyncReplicas);
+          int numReplicas = partitionInfo.replicas().length;
+          Node[] replicas = Arrays.copyOf(partitionInfo.replicas(), numReplicas);
+          Arrays.sort(replicas);
+          boolean isURP = !(replicas == inSyncReplicas);
```

Comment:
"==" looks not right here, as it compares the reference, not the array content. I think Arrays.equal should be used here. That said, I think we can extract a method to compare between ISR and replica set, and unit test it. That way we can feel safer before ship it.

Reply:
Yup, Array Equals makes more sense And Unit test aswell.